pickup/lib/s3.js

```diff
@@ -5,31 +5,17 @@ import { checkCar } from './car.js'
 import { logger } from './logger.js'
 
 /**
- * Upload CAR stream to temp bucket, verify it, then copy to destination bucket
+ * Verify CAR, by getting it back from the verification bucket and checking the dag is complete.
+ * If it's ok, calculate it's size and carCid and copy to destination bucket
  *
  * @param {object} config
  * @param {S3Client} config.client
  * @param {string} config.validationBucket
  * @param {string} config.destinationBucket
  * @param {string} config.key
- * @param {Readable} config.body
  * @param {string} config.cid
  */
-export async function uploadAndVerify ({ client, validationBucket, destinationBucket, key, body, cid }) {
-  // Handles s3 multipart uploading
-  // see: https://github.com/aws/aws-sdk-js-v3/blob/main/lib/lib-storage/README.md
-  const s3Upload = new Upload({
-    client,
-    params: {
-      Metadata: { structure: 'complete' },
-      Bucket: validationBucket,
-      Key: key,
-      Body: body
-    }
-  })
-
-  await s3Upload.done()
-
+export async function verify ({ client, validationBucket, destinationBucket, key, cid }) {
```

Comment:
move the uploading to the upload function and have verify just do verification. simples!